libp2p/switch.nim

```diff
@@ -166,6 +167,9 @@ proc upgradeOutgoing(s: Switch, conn: Connection): Future[Connection] {.async, g
     return
 
   result = await s.secure(result) # secure the connection
+  if isNil(result):
```

Comment:
so.. sometimes nil, sometimes exceptions here? because you only catch `CatchableError`, and not `Exception` in `secure`..

Reply:
You shouldn't handle `Exception` except for graceful shutdown. In this case all recoverable (`CatchableError`) exceptions are handled and non-recoverable (`Except`) are allowed to bubble up. Pretty standard stuff.